proto/cosmos/upgrade/v1beta1/query.proto

```diff
@@ -68,3 +73,19 @@ message QueryUpgradedConsensusStateResponse {
 
   bytes upgraded_consensus_state = 2;
 }
+
+// QueryVersionMap is the request type for the Query/VersionMap
+// RPC method.
+message QueryVersionMap {
+  // module_name is a field to query a specific module
+  // consensus version from state. Leaving this empty will
+  // fetch the full version map from state
+  string module_name = 1;
+}
+
+// QueryVersionMapResponse is the response type for the Query/VersionMap
+// RPC method.
+message QueryVersionMapResponse {
+  // version_map is a map of module name to consensus version.
+  map<string, uint64> version_map = 1;
```

Comment:
Hmm, i'm a bit wary of this. With ADR-033, Query services are part of the state machine, so I would like to avoid `map`s altogether (looping through them is not deterministic).

Could we do `repeated ModuleConsensusVersion`, where is `ModuleConsensusVersion` is a `message` with 2 fields string and uint64? And then this array would be sorted by module name (similarly to how we sort sdk.Coins).

Reply:
I think there ends up being a trade-off here in terms of efficiency vs. safety against the novice module developer. 

so if a module developer converts the map to an array for later storage, there is 100% chance that the next queried map won't have the same ordering. 

we do add some overhead to the query though, as we have to convert the map to a slice, and then alphabetically sort the slice by module name. (i think thats like O(nlogn*m). this may be negligible for a small query like this though. 

if we did go with returning a slice, it would probably be better to change the query name from `VersionMap` to something else (`ModuleConsensusVersions`?), since the underlying return type is no longer a `Map`.

curious if anyone else has thoughts - @aaronc ?

Reply:
I would love to use protobuf maps, and I think it's safer to just use a `repeated` field instead. Using protobuf maps in APIs is something that will require its whole own "safety study" or something.

Reply:
sounds good to me. the current commit on this PR has the `repeated` field implemented. That just leaves the issue of the naming:

`QueryVersionMap` returning a slice would be a bit odd imo. I am thinking we can either 
1. rename it
2. convert slice to map when request resolves in client 

thoughts @aaronc @amaurym ?

Reply:
Let's call it `QueryModuleVersions`

Reply:
`QueryModuleVersions` sounds good to me too